ocs_ci/ocs/longevity.py

```diff
@@ -0,0 +1,704 @@
+
+        return pvc_bound_list
+
+    def delete_stage_builder_kube_job(self, kube_job_obj_list, namespace):
+        """
+        Delete the stage builder kube jobs
+
+        Args:
+            kube_job_obj_list (list): List of kube jobs to delete
+            namespace (str): Namespace where the job is created
+
+        """
+        # Delete stage builder kube jobs
+        log.info("Deleting stage builder kube jobs ...")
+        for kube_job_obj in kube_job_obj_list:
+            kube_job_obj.delete(namespace=namespace)
+
+    def create_stagebuilder_all_pvc_types(
+        self, num_of_pvc, namespace, pvc_size, kube_job_name="all_pvc_job_profile"
+    ):
+        """
+        Create stagebuilder PVCs with all supported PVC types and access modes
+
+        Args:
+            num_of_pvc(int): Bulk PVC count
+            namespace (str): Namespace where the Kube job/PODs are to be created
+            pvc_size (str): size of all pvcs to be created with Gi suffix (e.g. 10Gi).
+            If None, random size pvc will be created
+
+        Returns:
+            pvc_bound_list (list): List of all PVCs in Bound state
+             pvc_job_file_list (list): List of all PVC.yaml dicts
+
+        """
+        log.info("Creating stagebuilder pods with all pvc types and access modes")
+        # Construct bulk PVC creation yaml for kube job
+        pvc_dict_list = self.construct_stage_builder_bulk_pvc_creation_yaml(
+            num_of_pvcs=num_of_pvc, pvc_size=pvc_size
+        )
+        # Construct kube job with the pvc dict list
+        pvc_job_file_list = self.construct_stage_builder_kube_job(
+            obj_dict_list=pvc_dict_list,
+            namespace=namespace,
+            kube_job_name=kube_job_name,
+        )
+        # Create stage builder for PVC kube job
+        self.create_stage_builder_kube_job(
+            kube_job_obj_list=pvc_job_file_list, namespace=namespace
+        )
+        # Validate PVCs in kube job reached BOUND state
+        pvc_bound_list = self.validate_pvc_in_kube_job_reached_bound_state(
+            kube_job_obj_list=pvc_job_file_list,
+            namespace=namespace,
+            pvc_count=num_of_pvc,
+        )
+
+        return pvc_bound_list, pvc_job_file_list
+
+    def construct_stage_builder_bulk_pod_creation_yaml(self, pvc_list, namespace):
+        """
+        This function constructs bulks pod.yamls to create bulk pods using kube_jobs.
+
+        Args:
+            pvc_list (list): List of PVCs
+            namespace (str): Namespace where the resource has to be created
+
+        Returns:
+            pods_dict_list (list): List of all Pod.yaml dict list
+
+        """
+        pods_dict_list = []
+        log.info("Constructing bulk pod creation yaml for the list of PVCs ")
+        # Get pvc objs from namespace
+        pvc_objs = get_pvc_objs(pvc_names=pvc_list, namespace=namespace)
+        for pvc_obj in pvc_objs:
+            if pvc_obj.backed_sc == constants.DEFAULT_STORAGECLASS_RBD:
```

Comment:
this doesn't work in an external-based cluster?

Reply:
Yes, external mode is out of scope for now. Also added @skipif_external_mode to stage0 and stage1 tests

Reply:
sorry, I missed it. Thanks :)